apps/src/javalab/JavalabEditor.jsx

```diff
@@ -410,6 +412,30 @@ class JavalabEditor extends React.Component {
     });
   }
 
+  onImportFile(filename, fileContents) {
+    const {fileMetadata} = this.state;
+    // If filename already exists in sources, replace file contents.
+    // Otherwise, create a new file.
+    if (Object.keys(this.props.sources).includes(filename)) {
+      // find editor for filename and overwrite contents of that editor
+      let editorKey = null;
+      for (const key in fileMetadata) {
+        if (fileMetadata[key] === filename) {
+          editorKey = key;
+        }
+      }
+      const editor = this.editors[editorKey];
+      editor.dispatch({
+        changes: {from: 0, to: editor.state.doc.length, insert: fileContents}
+      });
+      this.props.setSource(filename, fileContents);
```

Comment:
Curious - does `setSource` not overwrite the editor automatically?

Reply:
It doesn't. We have sort of a weird setup that could be looked at again post-pilot. The sources populate the editor, but then the editor controls its own state going forward (via codemirror) and sends updates back to sources for saving purposes. Before now we never put new contents into an existing editor after load.